src/Persistence/PersistenceManager.php

```diff
@@ -117,6 +122,29 @@ public function isPersistEnabled(): bool
         return true === $this->persistEnabled;
     }
 
+    /**
+     * @param class-string $targetClass
+     */
+    public static function classCanBePersisted(string $targetClass): bool
+    {
+        $reflectionClass = new \ReflectionClass($targetClass);
+
+        $attributes = $reflectionClass->getAttributes(ORMEntityAttribute::class);
+        if (!$attributes) {
+            $attributes = ($reflectionClass)->getAttributes(ODMDocumentAttribute::class);
+        }
+
+        if (!$attributes) {
+            $attributes = self::annotationReader()?->getClassAnnotation($reflectionClass, ORMEntityAttribute::class);
+        }
+
+        if (!$attributes) {
+            $attributes = self::annotationReader()?->getClassAnnotation($reflectionClass, ODMDocumentAttribute::class);
+        }
+
+        return (bool) $attributes;
+    }
```

Comment:
I know a better way to do this would have been to check doctrine's mapping, and check whether type is set as attribute or annotation, but this way I can check it statically